### PR TITLE
feat(source-stripe): Expands discounts and tax rates in stripe invoice stream (from: @aasimsani)

### PR DESCRIPTION
**Pull request by AGPapa**
## What
This is the same code as #37748, plus a change requested in the discussion on that PR. The PR seems stale, so I created this PR to address the feedback so we can publish this feature.

## How
The discounts column on invoices now gets an 'expand' option passed through, which turns the array of ids into an array of objects.

## Review guide
1. airbyte-integrations/connectors/source-stripe/source_stripe/source.py
2. airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json

## User Impact
Instead of seeing an array of discount ids, users will see the full discount object expanded as json.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json [hunk 1]
```diff
@@ -110,7 +110,69 @@
       "description": "Any discounts applied to the invoice",
       "type": ["null", "array"],
       "items": {
-        "type": ["null", "string"]
+        "type": ["null", "object"],
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json:113]:
```suggestion
        "type": ["null", "object"],
        "additionalProperties": true
```

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json [hunk 2]
```diff
@@ -110,7 +110,69 @@
...
+          },
+          "checkout_session": {
+            "type": ["null", "string"]
+          },
+          "coupon": {
+            "type": ["null", "object"],
+            "properties": {
+              "metadata": {
+                "type": ["null", "object"],
+                "properties": {}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json:129]:
```suggestion
                "additionalProperties": true,
                "properties": {}
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
I triggered CI tests to validate the change. It looks good, but we need final approval the connector team.

**Comment by marcosmarxm** [comment]
@airbytehq/critical-connectors can someone review this? Tests are green, need final review/approval from the team.

**Comment by bazarnov** [comment]
@ChristoGrab @natikgadzhi Could you please approve running the `regression tests` on this one?

**Comment by marcosmarxm** [comment]
Tests are passing. Need final approval from the team @airbytehq/dev-python

**Review by bazarnov — APPROVED** [review]

**Comment by AGPapa** [comment]
I also would like to expand the tax_rate on the same stream. I considered opening up a new PR for that, but I thought that it might be easier if it was part of this PR (so the version only needs to be upgraded once). Let me know if that's incorrect and I'd be happy to revert the commit and create a new PR to expand the tax_rate

File: airbyte-integrations/connectors/source-stripe/metadata.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: e094cb9a-26de-4645-8761-65c0c425d1de
-  dockerImageTag: 5.4.9
```
Comment (AGPapa) [inline airbyte-integrations/connectors/source-stripe/metadata.yaml:13]: Is this a patch version upgrade? Or should it be something else?

I just took that from the original PR, but let me know if this should be some other kind of version change instead.

Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/metadata.yaml:13]: Yes, it's considered as `patch`.

**Review by marcosmarxm — APPROVED** [review]

**Comment by aasimsani** [comment]
@marcosmarxm - I'm trying to use the latest version of the AirByte connector on my AirByte Cloud - would you happen to know how long it'd take for that to get updated?

**Pull request merged**